src/tests/system/tests/test_netgroups.py

```diff
@@ -108,3 +109,145 @@ def test_netgroups__add_remove_netgroup_member(client: Client, provider: Generic
+
+    for id in [1, 2]:
+        provider.user(f"ng{id}").add()
+
+    netgroup_qa = provider.netgroup("QAUsers").add()
+    netgroup_qa.add_member(host="testhost1", user="ng1", domain="ldap.test")
+
+    netgroup_dev = provider.netgroup("DEVUsers").add()
+    netgroup_dev.add_member(host="testhost5", user="ng2", domain="ldap.test")
+    if Operation == "Replace":
+        netgroup_dev.add_member(ng=netgroup_qa.dn)
+    else:
+        netgroup_dev.add_member(ng="QAUsers")
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("DEVUsers").members
+    assert "(testhost5, ng2, ldap.test)" in member
+    assert "(testhost1, ng1, ldap.test)" in member
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_netgroup__lookup_nested_groups(client: Client, provider: GenericProvider):
+    """
+    :title: Nesting netgroups and verifying user memberships using LDAP with sssd.
+    :setup:
+        1. Create users, groups.
+        2. Create netgroup named netgroup and Add Member
+        3. Create another netgroup named nested_netgroup
+        4. Add Members to nested_netgroup
+        5. Add Circular Netgroup Nesting to nested_netgroup
+        6. Start sssd
+    :steps:
+        1. Retrieves all members of the "nested_netgroup" group using the getent netgroup tool.
+        2. Verify that users from another group is also part of "nested_netgroup".
+        3. Checks if a user who is not in any netgroup is part of "nested_netgroup".
+        4. After the SSSD restart, it retrieves the members of "nested_netgroup" again to ensure they still intact.
+    :expectedresults:
+        1. All members of the "nested_netgroup" group be there
+        2. Users from another group is also part of "nested_netgroup".
+        3. User who is not in any netgroup is part of "nested_netgroup".
+        4. After restart all members of the "nested_netgroup" group be there
+    """
+    if isinstance(provider, IPA):
+        pytest.skip(reason="Not for IPA povider")
+
+    for id in [1, 2, 3]:
+        provider.user(f"ng{id}").add()
+
+    netgroup = provider.netgroup("netgroup").add()
+    netgroup.add_member(host="testhost1", user="ng1", domain="ldap.test")
+
+    nested_netgroup = provider.netgroup("nested_netgroup").add()
+    nested_netgroup.add_member(ng=netgroup.dn)
+    nested_netgroup.add_member(host="testhost5", user="ng2", domain="ldap.test")
+    nested_netgroup.add_member(user="ng3")
+
+    netgroup.add_member(ng=nested_netgroup.dn)
+
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("nested_netgroup").members
+    assert "(testhost1,ng1,ldap.test)" in member
+    assert "(-,ng3,)" in member
+    assert "(testhost5,ng2,ldap.test)" in member
+
+    client.sssd.restart()
+
+    member = client.tools.getent.netgroup("nested_netgroup").members
+    assert "(testhost1,ng1,ldap.test)" in member
+    assert "(-,ng3,)" in member
+    assert "(testhost5,ng2,ldap.test)" in member
```

Comment:
Why is this needed? Are you expecting some change may happen?

Reply:
No, this is assertion to make sure everything works

Reply:
What do you mean by everything? What is the exact use case you are trying to test?

Reply:
After sssd restart i want to make sure that Nesting netgroups works and i can remove that part if it does not make sense .

Reply:
IMO it doesn't. @danlavu what do you think? Does it make sense to check again the nesting groups after restarting sssd?